sycl/source/detail/device_impl.cpp

```diff
@@ -77,7 +77,7 @@ device_impl::device_impl(pi_native_handle InteropDeviceHandle,
 }
 
 device_impl::~device_impl() {
-  if (!MIsHostDevice) {
+  if (!MIsRootDevice && !MIsHostDevice) {
```

Comment:
@romanovvlad , I am reverting this change I made in PR 2232 because it caused this change caused an issue.
The problem is that somehow SYCL RT calls piDeviceRelease() for the same root device twice even if it is already released.
As a result, it caused pi_device cache behave weird and lots of LIT tests to fail.

Reply:
The https://github.com/intel/llvm/pull/2232 does not seem related, what's the real PR #? Also what is the plane to fix what that was fixing?

Reply:
Sorry PR 2256. https://github.com/intel/llvm/pull/2256
I need more time to understand why the previous change caused SYCL RT to call piDeviceRelease() for already released device.